### PR TITLE
feat(notification): GET Notification details endpoint

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

This PR intends to implements endpoint to get one notification details

**Fixes** # MON-19631

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Notification/Application/UseCase/FindNotification/FindNotification.php [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+                $notificationResources = $this->findResourcesByNotificationId($notificationId);
+
+                $presenter->presentResponse($this->createResponse(
+                    $notification,
+                    $notificationMessages,
+                    $notifiedUsers,
+                    $notificationResources
+                ));
+            }
+        } catch (\Throwable $ex) {
```
Comment (jeremyjaouen) [inline centreon/src/Core/Notification/Application/UseCase/FindNotification/FindNotification.php:98]: add catch for assertionfailed

File: centreon/src/Core/Notification/Application/UseCase/FindNotification/FindNotification.php [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        $response->users = array_map(
+            static fn(ConfigurationUser $user): array => ['id' => $user->getId(), 'name' => $user->getName()],
+            $notifiedUsers
+        );
+
+        foreach ($notificationResources as $resource) {
+            $responseResource = [
+                'type' => $resource->getType(),
+                'events' => $resource->getType() === NotificationResource::HOSTGROUP_RESOURCE_TYPE
+                    ? $response->convertHostEventsToBitFlags($resource->getEvents())
```
Comment (jeremyjaouen) [inline centreon/src/Core/Notification/Application/UseCase/FindNotification/FindNotification.php:175]: Convert into presenter

File: centreon/src/Core/Notification/Application/UseCase/FindNotification/FindNotificationResponse.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+     *
+     * @return int
+     */
+    public function convertServiceEventsToBitFlags(array $enums): int
+    {
+        /**
+         * @var NotificationServiceEvent[] $enums
+         */
+        return NotificationServiceEventConverter::toBitFlags($enums);
+    }
```
Comment (jeremyjaouen) [inline centreon/src/Core/Notification/Application/UseCase/FindNotification/FindNotificationResponse.php:96]: move to infra

File: centreon/doc/API/v23.10/Configuration/Notification/Notification.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+get:
+  tags:
+    - Notification
+  summary: "Get a notification details"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Notification.yaml:4]:
```suggestion
  summary: "Get the details of a notification"
```

File: centreon/doc/API/v23.10/Configuration/Notification/Notification.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  description: |
+    Get a notification details
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Notification.yaml:6]:
```suggestion
    Get the details of a notification
```

File: centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml [hunk 6]
```diff
@@ -0,0 +1,84 @@
+type: object
+properties:
+  id:
+    type: integer
+    description: "id of notification"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml:5]:
```suggestion
    description: "ID of the notification"
```

File: centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  name:
+    type: string
+    description: "Name of notification"
+    example: "notification-name"
+  timeperiod:
+    type: object
+    properties:
+      id:
+        type: integer
+        description: "id of the timeperiod"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml:16]:
```suggestion
        description: "ID of the timeperiod"
```

File: centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        example: 1
+      name:
+        type: string
+        description: "Name of timeperiod"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml:20]:
```suggestion
        description: "Name of the timeperiod"
```

File: centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    description: "notification status"
+    example: true
+  messages:
+    type: array
+    items:
+      type: object
+      properties:
+        channel:
+          type: string
+          description: "Channel use to send notification"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml:33]:
```suggestion
          description: "Channel used to send the notification"
```

File: centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+          type: string
+          description: "content of the message"
+  users:
+    type: array
+    items:
+      type: object
+      properties:
+        id:
+          type: integer
+          description: "id of the user"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml:47]:
```suggestion
          description: "ID of the user"
```

File: centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+          type: string
+          description: "name of the user"
+  resources:
+    type: array
+    items:
+      type: object
+      properties:
+        type:
+          type: string
+          description: "Type of Resource"
```
Comment (cg-tw) [inline centreon/doc/API/v23.10/Configuration/Notification/Schema/Notification.Details.yaml:58]:
```suggestion
          description: "Type of resource"
```

File: centreon/src/Core/Notification/Application/Exception/NotificationException.php [hunk 12]
```diff
@@ -71,4 +71,9 @@ public static function invalidUsers(int $notificationId): self
     {
         return new self(sprintf(_('Notification #%d should have at least one user'), $notificationId));
     }
+
+    public static function listOneNotAllowed(): self
+    {
+        return new self(_('You are not allowed to display notification details'));
```
Comment (cg-tw) [inline centreon/src/Core/Notification/Application/Exception/NotificationException.php:77]:
```suggestion
        return new self(_('You are not allowed to display the details of the notification'));
```

File: centreon/tests/php/Core/Notification/Application/UseCase/FindNotification/FindNotificationTest.php [hunk 13]
```diff
@@ -0,0 +1,236 @@
...
+        $this->readAccessGroupRepository,
+    ))(1, $this->presenter);
+
+    expect($this->presenter->responseStatus)
+        ->toBeInstanceOf(ErrorResponse::class)
+        ->and($this->presenter->responseStatus?->getMessage())
+        ->toBe(NotificationException::errorWhileRetrievingObject()->getMessage());
+});
+
+it('should get the resources with ACL Calculation when the user is not admin', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/Notification/Application/UseCase/FindNotification/FindNotificationTest.php:121]:
```suggestion
it('should get the resources with ACL calculation when the user is not admin', function (): void {
```

File: centreon/tests/php/Core/Notification/Application/UseCase/FindNotification/FindNotificationTest.php [hunk 14]
```diff
@@ -0,0 +1,236 @@
...
+
+    (new FindNotification(
+        $this->notificationRepository,
+        $contact,
+        $this->repositoryProvider,
+        $this->readAccessGroupRepository,
+    ))(1, $this->presenter);
+});
+
+it('should present a FindNotificationResponse when everything is ok', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/Notification/Application/UseCase/FindNotification/FindNotificationTest.php:164]:
```suggestion
it('should present a FindNotificationResponse when everything is OK', function (): void {
```

**Review by cg-tw — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Review by sc979 — APPROVED** [review]
Backport latest fix

**Pull request merged**